src/hdf5/wsfm_2ioda.py

```diff
@@ -0,0 +1,310 @@
+
+    k = 'brightnessTemperature'
+    VarAttrs[(k, 'ObsValue')]['_FillValue'] = float_missing_value
+    VarAttrs[(k, 'ObsError')]['_FillValue'] = float_missing_value
+    VarAttrs[(k, 'PreQC')]['_FillValue'] = int_missing_value
+    VarAttrs[(k, 'ObsValue')]['units'] = 'K'
+    VarAttrs[(k, 'ObsError')]['units'] = 'K'
+    # VarAttrs[(k, 'PreQC')]['units'] = 'unitless'
+
+    # final write to IODA file
+    writer.BuildIoda(obs_data, VarDims, VarAttrs, GlobalAttrs)
+
+
+def get_data_from_files(afile, skip=1):
+
+    # allocate space for output depending on which variables are to be saved
+    obs_data = init_obs_loc()
+
+    f = h5py.File(afile, 'r')
+    obs_data = get_data(f, obs_data, skip=skip)
+    f.close()
+
+    return obs_data
+
+
+def get_data(f, obs_data, skip=1):
+
+    WMO_sat_ID = get_WMO_satellite_ID(f.attrs['GranuleFilename'].decode("utf-8"))
+
+    nscans = f['NumScanR'][0]
+    nbeam_pos = f['NumFOVR'][0]
+    nchans = len(f['ChanFrequency'][:])
+    nbands = len(f['BandFrequency'][:])  # not used
+    band_norm = 2  # define a band to use for angles
+    obs_data[('latitude', metaDataName)] = np.array(f['Latitude'][:, :].flatten(), dtype='float32')
+    obs_data[('longitude', metaDataName)] = np.array(f['Longitude'][:, :].flatten(), dtype='float32')
+    obs_data[('longitude', metaDataName)][obs_data[('longitude', metaDataName)] > 180] -= 360.
+    obs_data[('sensorChannelNumber', metaDataName)] = np.array(np.arange(nchans)+1, dtype='int32')
+    obs_data[('sensorScanPosition', metaDataName)] = np.tile(np.arange(nbeam_pos, dtype='int32')+1, (nscans, 1)).flatten()
+    obs_data[('solarZenithAngle', metaDataName)] = np.array(f['SolarZenithAngle'][:, :].flatten(), dtype='float32')
+    obs_data[('solarAzimuthAngle', metaDataName)] = np.array(f['SolarAzimuthAngle'][:, :].flatten(), dtype='float32')
+    obs_data[('sensorZenithAngle', metaDataName)] = np.array(f['EarthIncidenceAngle'][band_norm, :, :].flatten(), dtype='float32')
+    obs_data[('sensorAzimuthAngle', metaDataName)] = np.array(f['EarthAzimuthAngle'][band_norm, :, :].flatten(), dtype='float32')
+    obs_data[('sensorViewAngle', metaDataName)] = np.array(f['EarthIncidenceAngle'][band_norm, :, :].flatten(), dtype='float32')
+
+    nlocs = len(obs_data[('latitude', metaDataName)])
+    obs_data[('satelliteIdentifier', metaDataName)] = np.full((nlocs), WMO_sat_ID, dtype='int32')
+
+    obs_data[('dateTime', metaDataName)] = np.array(np.repeat(f['ScanStartTime'][:] - et_offset, nbeam_pos), dtype='int64')
+
+    k = 'brightnessTemperature'
+    # have to reorder the channel axis to be last then merge ( nscans x nspots = nlocs )
+    # 'TbLowRes' - 'Brightness temperatures on the resampled scan grid for each polarimetric channel, 30 km RFOV'
+    # 'TbHighRes' - 'Brightness temperatures on the resampled scan grid for each polarimetric channel, 20 km RFOV'
+    # for channels less than 20 GHz use the TbLowRes and the TbHighRes for the other bands
+    freq_cf = 20.
```

Comment:
@nancylbaker here is the `crossover` frequency `freq_cf` values below this use the `TbLowRes` and frequencies above use the `TbHighRes`

let's discuss the way forward with this as I think it is largely ready to go and we can make smaller changes from there (assuming this is the type of file we want to decode)